tutorial/managing-software.md

```diff
@@ -0,0 +1,1394 @@
+versions, but it *could* end up removing some packages -- so although
+`apt upgrade` is safe to use unattended (in a script, for example), you should
+only use `dist-upgrade` when you can pay attention to it.
+
+### Searching APT
+
+Now we're up-to-date, we can start exploring! As with any other database, we
+can search APT to find software. Let's say that we want to find a webserver,
+for example. We can run the following command:
+
+```bash
+apt search webserver
+```
+
+This will return us a long list of all "webserver" packages it can find. But
+some of the descriptions don't actually contain the text "webserver" -- like in
+this section of the list:
+
+```text
+inotify-tools/noble 3.22.6.0-4 amd64
+  command-line programs providing a simple interface to inotify
+
+ipcalc/noble 0.51-1 all
+  parameter calculator for IPv4 addresses
+
+iwatch/noble 0.2.2-10 all
+  realtime filesystem monitoring program using inotify
+```
+
+We can use `apt show` to inspect the description and summary details of any
+package, so let's take a closer look at `ipcalc` from our list:
+
+```bash
+apt show ipcalc
+```
+
+The summary has been replaced with `[...]` for brevity, but we can see that the
+text "webserver" is in the long description of the "Description" field.
+
+```text
+Package: ipcalc
+Version: 0.51-1
+[...]
+APT-Sources: http://archive.ubuntu.com/ubuntu noble/universe amd64 Packages
+Description: parameter calculator for IPv4 addresses
+ ipcalc takes an IPv4 address and netmask and calculates the resulting
+ broadcast, network, Cisco wildcard mask, and host range. By giving a
+ second netmask, you can design sub- and supernetworks. It is also
+ intended to be a teaching tool and presents the results as
+ easy-to-understand binary values.
+ .
+ Originally, ipcalc was intended for use from the shell prompt, but a
+ CGI wrapper is provided to enable colorful HTML display through a
+ webserver.
+ You can find it in /usr/share/doc/ipcalc/examples directory.
+```
+
+In many places, you will see reference to `apt-get` and `apt-cache` instead of
+`apt`. Historically, the *database* part of APT was accessed using `apt-cache`
+(e.g. `apt-cache show ipcalc`), and the *packages* part of APT used `apt-get`
+(e.g. `apt-get install ipcalc`).
+
+APT has recently been streamlined, so although it uses `apt-get` and
+`apt-cache` "behind the scenes", you don't need to worry about remembering
+which command to use -- you can use `apt` directly. To find out more,
+refer to the manual pages for these commands. Type `man apt`, `man apt-get` or
+`man apt-cache` in your terminal to access them.
+
+## Installing deb packages
+
+For the following examples, we're going to use two popular webserver packages:
+Apache2 and nginx. However, we won't install anything just yet -- let's just
+take a look at the commands involved to understand what's happening.
+
+Installing deb packages using APT is done using the `apt install` command. We
+can install a single package:
```

Comment:
This part (implicitly) guides the user to actually run this command though. But if they install Apache here, then they won't be able to follow the next section, which assumes they don't have Apache installed yet.

Can you pick a different package for these two examples with apache?